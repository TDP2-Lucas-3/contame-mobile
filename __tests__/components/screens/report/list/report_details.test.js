import React from 'react';
import {shallow} from 'enzyme';
import ReportDetails from '../../../../../src/components/screens/reports/list/report_details';
import {findByTestId} from '../../../../support/helpers';

describe('ReportDetails', () => {
  let render;
  let props;

  beforeEach(() => {
    props = {
      title: 'a report',
      description: 'some description',
      images: ['some image data'],
    };
    render = (appProps) => shallow(<ReportDetails {...appProps} />);
  });

  it('renders without crashing', () => {
    expect(() => render(props)).not.toThrowError();
  });

  it('renders all values correctly', () => {
    expect(render(props)).toMatchSnapshot();
  });

  describe('when there is only one image', () => {
    it('renders given image', () => {
      expect(findByTestId(render(props), 'card_image')).toHaveProp(
        'source',
        expect.objectContaining({uri: props.images[0]}),
      );
    });
  });

  describe('when there are multiple images', () => {
    beforeEach(() => {
      props.images = ['first image data', 'second image data'];
    });

    it('renders first image', () => {
      expect(findByTestId(render(props), 'card_image')).toHaveProp(
        'source',
        expect.objectContaining({uri: props.images[0]}),
      );
    });
  });

  describe('when there is no images', () => {
    beforeEach(() => {
      props.images = [];
    });

    it("doesn't render any image", () => {
      expect(findByTestId(render(props), 'card_image')).toHaveLength(0);
    });

    it('renders default icon', () => {
      expect(findByTestId(render(props), 'default_icon')).toHaveLength(1);
    });
  });
});
