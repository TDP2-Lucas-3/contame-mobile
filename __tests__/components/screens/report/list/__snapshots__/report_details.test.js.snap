// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportDetails renders all values correctly 1`] = `
<ForwardRef>
  <Themed.Card
    containerStyle={
      Object {
        "borderRadius": 10,
        "borderWidth": 0,
        "elevation": 5,
        "padding": 0,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.34,
        "shadowRadius": 6.27,
      }
    }
  >
    <ReportImage
      image="some image data"
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1F271B",
          "borderBottomColor": "white",
          "borderBottomWidth": 1,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "margin": 10,
          }
        }
      >
        <Themed.Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Raleway-Bold",
              "fontSize": 16,
            }
          }
        >
          a report
        </Themed.Text>
        <Themed.Text
          style={
            Array [
              Object {
                "color": "grey",
              },
              Object {
                "fontFamily": "Raleway-Regular",
              },
              Object {
                "fontSize": 14,
              },
            ]
          }
        >
          a few seconds ago
        </Themed.Text>
      </View>
      <View
        style={
          Object {
            "margin": 10,
          }
        }
      >
        <Themed.Icon
          color="white"
          name="chevron-right"
          size={30}
        />
      </View>
    </View>
    <ReportCardBottomBar />
  </Themed.Card>
</ForwardRef>
`;
